app/controllers/medicaid/insurance_current_type_controller.rb

```diff
@@ -1,15 +1,80 @@
 # frozen_string_literal: true
 
 module Medicaid
-  class InsuranceCurrentTypeController < Medicaid::ManyMemberStepsController
+  class InsuranceCurrentTypeController < MedicaidStepsController
+    helper_method :current_member
+
+    def update
+      @step = step_class.new(step_params)
+
+      if @step.valid?
+        current_member.update(step_params)
+        redirect_to(next_path)
+      else
+        render :edit
+      end
+    end
+
+    def current_member
+      @_current_member ||= begin
+                             member_from_form ||
+                               member_from_querystring ||
+                               first_insurance_holder
+                           end
+    end
+
     private
 
+    def next_path
+      next_member_path || super
+    end
+
+    def next_member_path
+      return if next_member.nil?
+
+      decoded_step_path(params: { member: next_member.id })
+    end
+
+    def first_insurance_holder
+      current_application.
+        members.
+        insured.
+        first_member
+    end
+
+    def next_member
+      return if current_member.nil?
+
+      current_application.
+        members.
+        insured.
+        after(current_member).
+        first_member
+    end
+
     def skip?
       current_application.nobody_insured?
     end
 
     def member_attrs
       %i[insurance_type]
     end
+
+    def existing_attributes
+      HashWithIndifferentAccess.new(current_member&.attributes)
+    end
+
+    def member_from_form
+      return if params.dig(:step, :member_id).blank?
+
+      member_id = params[:step].delete(:member_id)
+      current_application.members.find(member_id)
+    end
+
+    def member_from_querystring
```

Comment:
when would this be used?

Reply:
When redirecting to the "next" member. see `next_path`

Reply:
gotcha! so we are using the querystring rather than nested routes. what is the thinking behind that?

Reply:
I don't think we could do nested routes considering we're automatically generating the routes via `Medicaid::StepNavigation`. We'd have to re-think the entire method to generate the routes if some of them turned from one controller and resource (the step) to two controllers: the parent question (in this case `type`) and the child (the member).